lib/ansible/modules/network/fortios/fortios_wireless_controller_wtp_profile.py

```diff
@@ -0,0 +1,1820 @@
+        split-tunneling-acl-path: "tunnel"
+        tun-mtu-downlink: "164"
+        tun-mtu-uplink: "165"
+        wan-port-mode: "wan-lan"
+'''
+
+RETURN = '''
+build:
+  description: Build number of the fortigate image
+  returned: always
+  type: str
+  sample: '1547'
+http_method:
+  description: Last method used to provision the content into FortiGate
+  returned: always
+  type: str
+  sample: 'PUT'
+http_status:
+  description: Last result given by FortiGate on last operation applied
+  returned: always
+  type: str
+  sample: "200"
+mkey:
+  description: Master key (id) used in the last call to FortiGate
+  returned: success
+  type: str
+  sample: "id"
+name:
+  description: Name of the table used to fulfill the request
+  returned: always
+  type: str
+  sample: "urlfilter"
+path:
+  description: Path of the table used to fulfill the request
+  returned: always
+  type: str
+  sample: "webfilter"
+revision:
+  description: Internal revision number
+  returned: always
+  type: str
+  sample: "17.0.2.10658"
+serial:
+  description: Serial number of the unit
+  returned: always
+  type: str
+  sample: "FGVMEVYYQT3AB5352"
+status:
+  description: Indication of the operation's result
+  returned: always
+  type: str
+  sample: "success"
+vdom:
+  description: Virtual domain used
+  returned: always
+  type: str
+  sample: "root"
+version:
+  description: Version of the FortiGate
+  returned: always
+  type: str
+  sample: "v5.6.3"
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+fos = None
+
+
+def login(data):
+    host = data['host']
+    username = data['username']
+    password = data['password']
+
+    fos.debug('on')
+    if 'https' in data and not data['https']:
+        fos.https('off')
+    else:
+        fos.https('on')
+
+    fos.login(host, username, password)
+
+
+def filter_wireless_controller_wtp_profile_data(json):
+    option_list = ['allowaccess', 'ap-country', 'ble-profile',
+                   'comment', 'control-message-offload', 'deny-mac-list',
+                   'dtls-in-kernel', 'dtls-policy', 'energy-efficient-ethernet',
+                   'ext-info-enable', 'handoff-roaming', 'handoff-rssi',
+                   'handoff-sta-thresh', 'ip-fragment-preventing', 'lan',
+                   'lbs', 'led-schedules', 'led-state',
+                   'lldp', 'login-passwd', 'login-passwd-change',
+                   'max-clients', 'name', 'platform',
+                   'poe-mode', 'radio-1', 'radio-2',
+                   'split-tunneling-acl', 'split-tunneling-acl-local-ap-subnet', 'split-tunneling-acl-path',
+                   'tun-mtu-downlink', 'tun-mtu-uplink', 'wan-port-mode']
+    dictionary = {}
+
+    for attribute in option_list:
+        if attribute in json and json[attribute] is not None:
+            dictionary[attribute] = json[attribute]
+
+    return dictionary
+
+
+def flatten_multilists_attributes(data):
+    multilist_attrs = []
+
+    for attr in multilist_attrs:
```

Comment:
Seems like this is trying to iterate through an empty list. Could you please clarify?

Reply:
Yes, we did this intentionally to support different versions of FortiGate. Depending on the version there can be some attributes that change from one version to another. If these attributes end up being "multioptions" they need to be processed by the method "flatten_multilists_attributes" before they are sent to fortiosapi. The user can add the attributes to the list and this way we avoid generating another ansible module for each different version of fortigate.

Reply:
Instead of having the user modify the code, can we have a parameter for the modules where user can pass these attributes and map this variable `multilist_attrs` to this parameter?

Reply:
Thanks Nilashish for your comment. We thought about that option but in that case we would be impacting the parameters in the module and we would like to have a clean and clear module specification so that the regular user does not get confused about it. In this way the user will see only parameters and attributes that fortigate can understand and we won't be mixing fortigate configuration with ansible configuration. If it is too much trouble I could remove the code but if possible I would like to leave it there.